components/ColorModeToggleSwitch.tsx

```diff
@@ -0,0 +1,44 @@
+import { styled, Switch } from '@mui/material';
+
+export const ColorModeToggleSwitch = styled(Switch)(({ theme }) => ({
```

Comment:
This is mostly just brought in from the MUI Switch docs here: https://mui.com/material-ui/react-switch/#customization
I've tweaked the colours slightly and extracted the svgs into the public folder rather than inlining them.